http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HTTPAppDescriptorDAO.scala

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package dao
+
+import cats.effect.Effect
+import cats.implicits._
+import cats.mtl.Ask
+import io.chrisdavenport.log4cats.StructuredLogger
+import io.circe.Decoder
+import io.circe.yaml.parser
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.leonardo.JsonCodec._
+import org.broadinstitute.dsde.workbench.leonardo.dao.HttpAppDescriptorDAO._
+import org.http4s.{Method, Request, Response, Uri}
+import org.http4s.client.Client
+
+class HttpAppDescriptorDAO[F[_]](httpClient: Client[F])(implicit logger: StructuredLogger[F], F: Effect[F])
+    extends AppDescriptorDAO[F] {
+  override def getDescriptor(path: String)(implicit ev: Ask[F, TraceId]): F[AppDescriptor] =
+    for {
+      _ <- logger.info(s"Attempting to retrieve descriptor $path")
+      yamlStr <- resolveUri(path)
+      _ <- logger.info(s"Attempting to parse descriptor for $path as a yaml string into JSON...")
+      json <- F.fromEither(parser.parse(yamlStr))
+      _ <- logger.info(s"Attempting to parse json for $path into AppDescriptor model...")
+      res <- F.fromEither(json.as[AppDescriptor])
+      _ <- logger.info(s"Succesfully processed descriptor at $path")
+    } yield res
+
+  private[dao] def resolveUri(path: String)(implicit ev: Ask[F, TraceId]): F[String] =
+    for {
+      resp <- httpClient.expectOr[String](
+        Request[F](
+          method = Method.GET,
+          uri = Uri.unsafeFromString(path)
+        )
+      )(onError(path))
+    } yield resp
+
+  private def onError(path: String)(response: Response[F])(implicit ev: Ask[F, TraceId]): F[Throwable] =
+    for {
+      traceId <- ev.ask
+      body <- response.bodyText.compile.foldMonoid
+      _ <- logger.error(s"${traceId} | call failed: $body")
```

Comment:
Instead of embedding the trace id in the string, do this instead:

`logger.error(ctx.loggingCtx)(s"call failed: $body")`